### PR TITLE
Add missing SWIFT_FALLTHROUGH to the tryCastToString

### PR DESCRIPTION
**Pull request by shabalind**
This change fixes a warning in tryCastToString, that can break the build if `-Werror`, `-Wimplicit-fallthrough` flags are enabled:

```
swift/stdlib/public/runtime/DynamicCast.cpp:866:3: error: unannotated fall-through between switch labels [-Werror,-Wimplicit-fallthrough]
  default:
```

### CODE REVIEW

**Review by saeta — APPROVED** [review]
LGTM

**Comment by shabalind** [comment]
@swift-ci please test

File: stdlib/public/runtime/DynamicCast.cpp [hunk 1]
```diff
@@ -863,6 +863,7 @@ tryCastToString(
       takeOnSuccess, mayDeferChecks);
 #endif
   }
+    SWIFT_FALLTHROUGH;
```
Comment (xwu) [inline stdlib/public/runtime/DynamicCast.cpp:866]: Nit: maybe move this one line up, inside the braces?

Comment (shabalind) [reply stdlib/public/runtime/DynamicCast.cpp:865]: Moved in the follow-up commit.

**Comment by xwu** [comment]
@swift-ci please test

**Comment by xwu** [comment]
@shabalind Looks like this change introduces an unreachable code warning on platforms with Obj-C interop. I've followed up with #33912.

**Pull request merged**